packages/renderer/src/lib/donut/Donut.svelte

```diff
@@ -21,21 +22,40 @@ function describeArc(radius: number, endAngle: number) {
   return ['M', start.x, start.y, 'A', radius, radius, 0, largeArcFlag, 0, radius, 0].join(' ');
 }
 
-$: stroke = percent < 0 ? '' : percent < 50 ? 'stroke-green-500' : percent < 75 ? 'stroke-amber-500' : 'stroke-red-500';
+function getStroke(loading: boolean, percent: number): string {
+  if (loading) return 'stroke-purple-500';
 
-$: tooltip = percent ? percent.toFixed(0) + '% ' + title + ' usage' : '';
+  if (percent < 0) return '';
+
+  if (percent < 50) return 'stroke-green-500';
+
+  if (percent < 75) return 'stroke-amber-500';
+
+  return 'stroke-red-500';
+}
+
+function getShape(): string {
+  if (loading) return describeArc(size / 2, 72); // 20%
+  return describeArc(size / 2, (percent * 360) / 100);
+}
+
+function getTooltip(loading: boolean, title: string, percent: number): string {
+  if (loading) return `Loading ${title}...`;
+  return percent ? percent.toFixed(0) + '% ' + title + ' usage' : '';
+}
+
+$: stroke = getStroke(loading, percent);
+
+$: tooltip = getTooltip(loading, title, percent);
 </script>
 
 <Tooltip tip="{tooltip}" bottom>
   <svg viewBox="-4 -4 {size + 8} {size + 8}" height="{size}" width="{size}">
-    <circle fill="none" class="stroke-charcoal-300" stroke-width="1" r="{size / 2}" cx="{size / 2}" cy="{size / 2}"
-    ></circle>
-    <path
-      fill="none"
-      class="{stroke}"
-      stroke-width="3.5"
-      d="{describeArc(size / 2, (percent * 360) / 100)}"
-      data-testid="arc"></path>
+    <g class:animate-spin="{loading}" class:origin-[50%_50%]="{loading}" style="transform-box: fill-box;">
```

Comment:
the `transform-box: fill-box;` is not yet supported by tailwinds, ref https://github.com/tailwindlabs/tailwindcss/pull/10982